lib/archethic/replication/transaction_validator.ex

```diff
@@ -63,12 +64,25 @@ defmodule Archethic.Replication.TransactionValidator do
         inputs,
         self_repair?
       ) do
-    with :ok <- valid_transaction(tx, inputs, true, self_repair?),
+    with :ok <- validate_pending_transaction(tx, self_repair?),
```

Comment:
As the flag will be removed, I think you can omit the self-repair condition in this function.

Reply:
Yes the self repair flag is not used anymore here